akka-remote/src/main/scala/akka/remote/artery/compress/OutboundActorRefCompression.scala

```diff
@@ -101,6 +107,39 @@ private[remote] class OutboundCompressionTable[T](system: ActorSystem, remoteAdd
 
 }
 
+class HashScanIndexMap[T](val size: Int) {
```

Comment:
IMO that's pretty much the fastest structure we can use for these things.
The assumption is that since ids start from 0, and are required to be gapless we can just sort + put them into an array, where which emulates a map by "ref == key", "index == value"